### PR TITLE
SyncActivity | Bluetooth

### PR DESCRIPTION
**Pull request by LinX64**

### CODE REVIEW

**Comment by LinX64** [comment]
[Fatal Exception: java.lang.RuntimeException](https://trello.com/c/Ue9eV9fu/1664-fatal-exception-javalangruntimeexception)

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/sync/SyncController.kt [hunk 1]
```diff
@@ -162,45 +173,62 @@ class SyncController(
...
+            when {
+                ContextCompat.checkSelfPermission(
+                    mRootActivity,
+                    Manifest.permission.BLUETOOTH_CONNECT
+                ) != PackageManager.PERMISSION_GRANTED -> {
+                    mPermissionsManager.requestBluetoothPermissions(mRootActivity)
+                }
+            }
+        }
+    }
```
Comment (xameYeoh) [inline app/src/main/java/pl/llp/aircasting/ui/view/screens/sync/SyncController.kt:187]: I think the right bluetooth manager is being injected here, so maybe we can have this request in its overridden method: 
```
override fun requestBluetoothEnable(activity: Activity?) {
        val intent = Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE)

        if (activity?.let {
                ActivityCompat.checkSelfPermission(
                    it,
                    Manifest.permission.BLUETOOTH_CONNECT
                )
            } == PackageManager.PERMISSION_GRANTED
        ) {
            activity.startActivityForResult(
                intent,
                ResultCodes.AIRCASTING_REQUEST_BLUETOOTH_ENABLE
            )
        }
    }
```

Comment (xameYeoh) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/sync/SyncController.kt:187]: So we can request it in the else block here I think

File: app/src/main/java/pl/llp/aircasting/util/helpers/bluetooth/BluetoothRuntimePermissionManager.kt [hunk 2]
```diff
@@ -54,30 +56,38 @@ open class BluetoothRuntimePermissionManager(
...
+                ActivityCompat.checkSelfPermission(it, Manifest.permission.BLUETOOTH_CONNECT)
+            } == PackageManager.PERMISSION_GRANTED) {
             activity.startActivityForResult(
                 intent,
                 ResultCodes.AIRCASTING_REQUEST_BLUETOOTH_ENABLE
             )
         }
     }
+
+    private fun needNewBluetoothPermissions(activity: Activity?) {
```
Comment (xameYeoh) [inline app/src/main/java/pl/llp/aircasting/util/helpers/bluetooth/BluetoothRuntimePermissionManager.kt:81]: How about we have a naming like "askForBluetoothConnectPermissionIfNotGranted()"?

Comment (xameYeoh) [reply app/src/main/java/pl/llp/aircasting/util/helpers/bluetooth/BluetoothRuntimePermissionManager.kt:81]: or requestForBluetoothConnectPermissionIfNotGranted()

**Pull request merged**